Sources/CodableKit/Reflection/HiLoEncoder.swift

```diff
@@ -1,135 +1,136 @@
 /// Encodes types, detecting coding path for "hi" signal values.
-struct HiLoEncoder: Encoder {
+struct HiLoEncoder<Root, Value>: Encoder {
     let codingPath: [CodingKey]
     let userInfo: [CodingUserInfoKey: Any] = [:]
     private let ctx: Context
-    
+
     var hi: [CodingKey]? {
         return ctx.hiCodingPath
     }
-    
-    init() {
+
+    init(keyPath: KeyPath<Root, Value>?) {
```

Comment:
and here

Reply:
That one is used on line 58 in Codable+Reflectable. But I can remove it and change the Decoder / Encoder to only take the the generic type to look for?

Reply:
I think it would be better to keep it consistent and use explicit generic (i.e., `<Root, Value>`) for init

Reply:
I had an idea to add `init(single: Signal, scanFor: Value.Type)` but you want `init<Value>(signal: Signal)`?

Reply:
We can remove the `Root` generic since this is never used